src/it/resources/com/google/checkstyle/test/chapter4formatting/rule412nonemptyblocks/InputRightCurlySwitchCasesBlocks.java

```diff
@@ -72,15 +78,100 @@ public static void test3() {
     int k = 0;
     switch (k) {
       case 1:
-        { // violation ''{' at column 9 should be on the previous line.'
+        {
           int x = 1;
           break;
         }
       case 2:
-        { // violation ''{' at column 9 should be on the previous line.'
+        {
         int x = 2; } // violation ''}' at column 20 should be alone on a line.'
         break;
       default:
+        {
+          int x = 2; } // false-negative
+    }
+  }
+
+  /** some javadoc. */
+  public static void test4() {
+    int mode = 0;
+    switch (mode) {
+      case 1: { // violation ''{' at column 15 should be on a new line.'
+          int x = 1; // violation '.* incorrect indentation level 10, expected level should be 8.'
+          break; // violation '.* incorrect indentation level 10, expected level should be 8.'
+        } // violation '.* incorrect indentation level 8, expected level should be 6.'
+      case 2: { // violation ''{' at column 15 should be on a new line.'
+          int x = 0; // violation '.* incorrect indentation level 10, expected level should be 8.'
+          break; // violation '.* incorrect indentation level 10, expected level should be 8.'
+        } // violation '.* incorrect indentation level 8, expected level should be 6.'
+      default: { // violation ''{' at column 16 should be on a new line.'
+        int x = 0;
+      }
+    }
+  }
+
+  /** some javadoc. */
+  public static void test5() {
```

Comment:
Looks like you added lots of similar use cases. Why do we need them? Could you keep unique ones and remove the rest?

Reply:
I could there's no problem for it. But I tried to cover edge cases which seems similar but have little differences like whitespaces, indentation, etc.

Reply:
I have removed the similar test cases and tried to keep only unique ones